src/container.jl

```diff
@@ -7,13 +7,7 @@ end
 const Particle = Trace
 
 function Trace(f, rng::TracedRNG)
-    ctask = let f = f
-        Libtask.CTask() do
-            res = f(rng)
-            Libtask.produce(nothing)
-            return res
-        end
-    end
+    ctask = Libtask.CTask(f, rng)
```

Comment:
Why not match the new Turing interface ? Or that's for a later PR ?
```julia
  if hasfield(typeof(f), :evaluator) # Test whether f is a Turing.TracedModel
      # println(f.evaluator)
      ctask = Libtask.CTask(f.evaluator[1], f.evaluator[2:end]...)
  else # f is a Function, or AdavncedPS.Model
      ctask = Libtask.CTask(f)
  end
```

Reply:
yes, we should match that - please feel free to update this PR!

Reply:
Couldn't/Shouldn't this be handled in Turing by implementing `Libtask.CTask(f::TracedModel, rng)`?